sentry_sdk/tracing.py

```diff
@@ -588,17 +590,34 @@ def finish(self, hub=None):
         # to be garbage collected
         self._span_recorder = None
 
-        return hub.capture_event(
-            {
-                "type": "transaction",
-                "transaction": self.name,
-                "contexts": {"trace": self.get_trace_context()},
-                "tags": self._tags,
-                "timestamp": self.timestamp,
-                "start_timestamp": self.start_timestamp,
-                "spans": finished_spans,
-            }
-        )
+        event = {
+            "type": "transaction",
+            "transaction": self.name,
+            "contexts": {"trace": self.get_trace_context()},
+            "tags": self._tags,
+            "timestamp": self.timestamp,
+            "start_timestamp": self.start_timestamp,
+            "spans": finished_spans,
+        }
+
+        if has_custom_measurements_enabled():
+            event["measurements"] = self._measurements
+
+        return hub.capture_event(event)
+
+    def set_measurement(self, name, value, unit="ms"):
+        # type: (str, float, MeasurementUnit) -> None
```

Comment:
is the type hinting preventing from using None as a name. Or: should we test name here for invalid names?

Reply:
I have no clue what types/mypy do here lol, I would kill them if you asked me 🤷